### PR TITLE
Feature/modified panoptic quality

### LINKED ISSUE #1591: Add Modified Panoptic Quality metric
## 🚀 Feature

Add Modified Panoptic Quality metric introduced in [Seamless Scene Segmentation](https://arxiv.org/abs/1905.01220), Section 4 "Revisiting Panoptic Segmentation".

### Motivation

The Modified Panoptic Quality is a subtle modification of the original, and it comes in handy to have it available when dealing with Panoptic Segmentation problems: it's not easy to say whether the original or the modified one is a better choice in any given situation, so one would ideally track both. 
To prove the point, some public datasets such as [nuScenes](https://www.nuscenes.org/panoptic/?externalData=all&mapData=all&modalities=Any) use both variants for evaluation on the leaderboard.

### Pitch

I'm happy to continue the work on PQ to extend its functionality 😉 

### Additional context

Question to the maintainers: would you rather see this implemented in its own class and functional versions, or as a switch (e.g., `implementation: Literal['original', 'modified'] = 'original'`) of the already existing metrics?
The code is almost identical besides a couple of extra `if` statements in the `update()` step, and so are the signatures, so I would lean on the latter option to avoid additional copy-paste.

### PR DESCRIPTION
**Pull request by marcocaccin**
## What does this PR do?

Closes #1591 

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Adds Modified Panoptic Quality as a separate metric, heavily borrowing common functionalities from the original implementation.

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/torchmetrics/detection/modified_panoptic_quality.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+from torchmetrics.utilities.plot import _AX_TYPE, _PLOT_OUT_TYPE
+
+if not _MATPLOTLIB_AVAILABLE:
+    __doctest_skip__ = ["ModifiedPanopticQuality.plot"]
+
+
+class ModifiedPanopticQuality(Metric):
+    r"""Compute `Modified Panoptic Quality`_ for panoptic segmentations.
+
+    The metric was introduced in https://arxiv.org/abs/1905.01220 Section 4, and is an adaptation of the original
```
Comment (SkafteNicki) [inline src/torchmetrics/detection/modified_panoptic_quality.py:39]:
```suggestion
    The metric was introduced in `Seamless Scene Segmentation paper`_, and is an adaptation of the original
```
please add https://arxiv.org/abs/1905.01220 to https://github.com/Lightning-AI/metrics/blob/master/docs/source/links.rst instead

File: src/torchmetrics/functional/detection/panoptic_quality.py [hunk 2]
```diff
@@ -41,6 +41,7 @@ def panoptic_quality(
     where IOU, TP, FP and FN are respectively the sum of the intersection over union for true positives, the number of
     true postitives, false positives and false negatives. This metric is inspired by the PQ implementation of
     panopticapi, a standard implementation for the PQ metric for object detection.
+    The metric was introduced in https://arxiv.org/abs/1801.00868.
```
Comment (SkafteNicki) [inline src/torchmetrics/functional/detection/panoptic_quality.py:44]:
```suggestion
    The metric was introduced in `Panoptic Segmentation paper`_.
```
again add https://arxiv.org/abs/1801.00868 to https://github.com/Lightning-AI/metrics/blob/master/docs/source/links.rst

**Review by SkafteNicki — APPROVED** [review]
LGTM :)

File: src/torchmetrics/detection/modified_panoptic_quality.py (hunk 1 above)
Comment (marcocaccin) [reply src/torchmetrics/detection/modified_panoptic_quality.py]: [68ac867](https://github.com/Lightning-AI/metrics/pull/1627/commits/68ac86737e34db0592ad73347640ca5978d08174)

**Review by Borda — APPROVED** [review]

**Pull request merged**